src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -96,22 +105,30 @@ interface IProps {
   setDateCallback?: (date: Date) => void
   setTimeCallback?: (date: Date) => void
   allowAfterMidnight?: boolean,
+  allowManualTimeChange?: boolean,
 }
 
 const DateTimeBlock : React.FC<IProps> = ({
   allowAfterMidnight = false,
   title,
   hasDate,
   hasTime,
+  allowManualTimeChange,
   date = new Date(),
   setDateCallback = ()=>{},
 }) => {
 
-
-  const [displayHours, setDisplayHours] = useState<string>(format(date, "mm"));
-  const [displayMinutes, setDisplayMinutes] = useState<string>(format(date,'HH'));
+  const convertHours = (date.getHours()).toString();
+  const convertMinutes = date.getMinutes().toString();
+  const [displayHours, setDisplayHours] = useState<string>(convertHours === '0'?'00':convertHours);
+  const [displayMinutes, setDisplayMinutes] = useState<string>(convertMinutes === '0'?'00':convertMinutes);
```

Comment:
We have similar logic being used to set values in `onBlurInputs`. Could be have a function that handles these cases and can be re-used more cleanly? At the moment the code repetition could be problematic if this behaviour is iterated again in the future.

Reply:
Done